buildSrc/src/main/java/quilt/internal/tasks/unpick/gen/OpenGlConstantUnpickGeneratorTask.java

```diff
@@ -0,0 +1,389 @@
+package quilt.internal.tasks.unpick.gen;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.PrintStream;
+import java.net.URI;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Predicate;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipFile;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.TextNode;
+import com.fasterxml.jackson.dataformat.xml.XmlMapper;
+import org.apache.commons.io.FileUtils;
+import org.gradle.api.file.RegularFileProperty;
+import org.gradle.api.provider.MapProperty;
+import org.gradle.api.tasks.Input;
+import org.gradle.api.tasks.InputFile;
+import org.gradle.api.tasks.OutputFile;
+import org.gradle.api.tasks.TaskAction;
+import org.objectweb.asm.ClassReader;
+import org.objectweb.asm.ClassVisitor;
+import org.objectweb.asm.FieldVisitor;
+import org.objectweb.asm.MethodVisitor;
+import org.objectweb.asm.Opcodes;
+import org.quiltmc.launchermeta.version.v1.DownloadableFile;
+import org.quiltmc.launchermeta.version.v1.Library;
+import org.quiltmc.launchermeta.version.v1.Version;
+import quilt.internal.Constants;
+import quilt.internal.tasks.DefaultMappingsTask;
+
+public abstract class OpenGlConstantUnpickGeneratorTask extends DefaultMappingsTask implements UnpickGen {
+    public static final String TASK_NAME = "openGlUnpickGen";
+    public static final String OPEN_GL_REGISTRY =
+        "https://raw.githubusercontent.com/KhronosGroup/OpenGL-Registry/main/xml/gl.xml";
+    public static final String GL_STATE_MANAGER_CLASS = "com/mojang/blaze3d/platform/GlStateManager";
+
+    public static final List<String> OPEN_GL_VERSIONS =
+        List.of("11", "12", "13", "14", "15", "20", "21", "30", "31", "32");
+
+    private static final Predicate<String> LWJGL_LIBRARY_PREDICATE =
+        Pattern.compile("^org\\.lwjgl:lwjgl-opengl:([^:]*)$").asMatchPredicate();
+
+    @InputFile
+    public abstract RegularFileProperty getVersionFile();
+
+    @InputFile
+    public abstract RegularFileProperty getPerVersionMappingsJar();
+
+    @Input
+    public abstract MapProperty<String, File> getArtifactsByUrl();
```

Comment:
I have a sneaking suspicion this is meant to be `@InputFiles`, if you're using the artifacts' contents.